### PR TITLE
dotnet54 stream changes

### PR DESCRIPTION
**Pull request by Yantrio**
part of #2220
ifdefed out old async stream overrides
replaced close -> dispose in dotnet54 stream usage
removed unused StreamExtensions.cs

### CODE REVIEW

**Comment by khellang** [comment]
Want to fix that history?

:watch:

**Comment by Yantrio** [comment]
fixed :watch:

**Comment by jchannon** [comment]
LGTM :watch: :eyes:

**Comment by thecodejunkie** [comment]
LGTM

File: src/Nancy/project.json [hunk 1]
```diff
@@ -10,11 +10,9 @@
     "compilationOptions": {
         "define": [ "DNX" ]
     },
-
```
Comment (khellang) [inline src/Nancy/project.json]: Can we please stop removing these? :disappointed:

Comment (Yantrio) [inline src/Nancy/project.json]: fixed, shouldn't happen again.

**Pull request merged**